features/lockscreen/impl/src/main/kotlin/io/element/android/features/lockscreen/impl/pin/model/PinEntry.kt

```diff
@@ -50,14 +50,36 @@ data class PinEntry(
         return copy(digits = newDigits.toPersistentList())
     }
 
+    fun deleteLast(): PinEntry {
+        if (isEmpty()) return this
+        val newDigits = digits.toMutableList()
+        newDigits.indexOfLast { it is PinDigit.Filled }.also { lastFilled ->
+            newDigits[lastFilled] = PinDigit.Empty
+        }
+        return copy(digits = newDigits.toPersistentList())
+    }
+
+    fun addDigit(digit: Char): PinEntry {
+        if (isComplete()) return this
+        val newDigits = digits.toMutableList()
+        newDigits.indexOfFirst { it is PinDigit.Empty }.also { firstEmpty ->
+            newDigits[firstEmpty] = PinDigit.Filled(digit)
+        }
+        return copy(digits = newDigits.toPersistentList())
+    }
+
```

Comment:
I think the method below `clear()` is not correct (but it's not used).
It should return `PinEntry.empty(size)`

Maybe adding unit test to cover `PinEntry` behavior could help.